src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -265,7 +271,26 @@ public IRecipeStorage getFirstFullFillableRecipe(final Predicate<ItemStack> stac
     public boolean fullFillRecipe(final IRecipeStorage storage)
     {
         final List<IItemHandler> handlers = getHandlers();
-        return storage.fullfillRecipe(this.getColony().getWorld(), handlers);
+
+        final AbstractEntityCitizen worker = this.getMainCitizen().getEntity().orElse(null);
+
+        if(worker == null)
+        {
+            // we shouldn't hit this case, but just in case...
+            return storage.fullfillRecipe(this.getColony().getWorld(), handlers);
```

Comment:
can't we just return false or so?

Reply:
This creates full backward compat, in case there is a scenario I missed, and the worker isn't set.